src/darwin/Framework/CHIP/CHIPP256KeypairBridge.mm

```diff
@@ -0,0 +1,143 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "CHIPP256KeypairBridge.h"
+
+#import <CommonCrypto/CommonDigest.h>
+#import <Security/SecKey.h>
+#include <string>
+
+#import "CHIPKeypair.h"
+#import "CHIPLogging.h"
+
+using namespace chip::Crypto;
+
+CHIP_ERROR CHIPP256KeypairBridge::Init(id<CHIPKeypair> keypair)
+{
+    mKeypair = keypair;
+    return setPubkey();
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::Initialize()
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return ([mKeypair initialize]) ? CHIP_NO_ERROR : CHIP_ERROR_INTERNAL;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::Serialize(P256SerializedKeypair & output) const
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::Deserialize(P256SerializedKeypair & input)
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::NewCertificateSigningRequest(uint8_t * csr, size_t & csr_length)
+{
+    if (!HasKeypair()) {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+}
+
+CHIP_ERROR CHIPP256KeypairBridge::ECDSA_sign_msg(const uint8_t * msg, size_t msg_length, P256ECDSASignature & out_signature)
+{
+    if (!HasKeypair()) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: no keypair to sign with.");
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+    NSData * msgData = [NSData dataWithBytes:msg length:msg_length];
+    NSMutableData * hashedData = [NSMutableData dataWithLength:CC_SHA256_DIGEST_LENGTH];
+    CC_SHA256(msgData.bytes, (CC_LONG) msgData.length, (unsigned char *) hashedData.mutableBytes);
+    CHIP_LOG_DEBUG("Generated Msg hash, signing hash now");
+    NSData * signature = [mKeypair ECDSA_sign_hash:hashedData];
+    if (!signature) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: no signature returned");
+        return CHIP_ERROR_INTERNAL;
+    }
+    if (signature.length > out_signature.Capacity()) {
+        CHIP_LOG_ERROR("ECDSA sign msg failure: unexpected signature size %tu vs %tu ", signature.length, out_signature.Capacity());
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    std::memmove(out_signature, signature.bytes, signature.length);
```

Comment:
since the move is signature.length, should we ensure that signature.length matches out_signature.Capacity exactly (rather than just <= check above)? Otehrwise we should have the signature.length in the output information as well.

supernit: Would memcpy be "more correct" since signature and out_signature memory regions can never overlap? 

Same to comments for the sign below.